### ISSUE TITLE
0.9.37.0 - Memory utilization for cross region job executions spiked

### ISSUE THREAD

**Issue by PratMis** [cloud-custodian/cloud-custodian#9558]
### Describe the bug

I'm not sure what exactly is causing it, but our memory utilization for custodian jobs running cross region, for example - Jobs executing on a cluster in us-east-1 but looking at resources in ca-central-1. Weird thing is that, this memory spike is only seen in busy accounts (potentially with more number of resources). For smaller accounts, we're seeing a dip as expected. Initial investigation, I found that the jobs that had an upper bound on allocated memory failed at the start of policy execution targeting `resource: s3`

For jobs running for us-east-1 on the cluster in us-east-1, memory didn't spike at all. For small accounts, we did see a dip for this scenario too
In these scenarios, the logs are shipped to a bucket in the region the cluster is hosted which is us-east-1
### What did you expect to happen?

Expected the memory utilization to go down. 
### Cloud Provider

Amazon Web Services (AWS)

### Relevant log/traceback output

```shell
Custodian version - 0.9.35
ca-central-1	cadence	11:24 05/30/2024	11:45 05/30/2024	00:20:43(duration)	0 (Success)	N/A	13914(allocated memory)	11192(memory utilized)	80.44 %(Used memory percent)	2(Allocated CPU)	49 %(Average Used CPU Percent)

Custodian version - 0.9.37 (Job failed due to OOM error)
ca-central-1	cadence	17:22 05/30/2024	17:41 05/30/2024	00:19:29	-9 (Essential exit)	N/A	13990	13990	100.00 %	1	71 %

We stabilized around 
ca-central-1	cadence	23:24 05/30/2024	23:47 05/30/2024	00:22:57	0 (Success)	N/A	24482	20030(memory utilized)	81.82 %	1	72 %

We jumped from 14G to 20G
```

**Comment by kapilt** [comment]
This is likely related to the client caching that went in 0.9.37.0 from this pr #9107

**Comment by kapilt** [comment]
I think the real challenge is we need to have a more definitive point to call close. 

is it a single run with a large list of policies? what's the delta on total time for the run? how long is the policy run.

**Comment by PratMis** [comment]
We see memory spike in both the scenarios found below (this is only big accounts)
1. Single runs with a large list of policies - One run contained 1210 policies, the other run contained 185 policies
2. Delta on total time for run - With 0.9.35.0 it took 22:44 minutes to run 1210 policies cross region (cluster in us-east-1 and policies run against ca-central-1). With 0.9.37.0 it took 00:24:07 minutes to run the same 1210 policies cross region.
For the run that entailed 185 policies it took 11:24min and 11:39min respectively for 0.9.35.0 and 0.9.37.0

Number of resources, especially s3 buckets - 937 in one of the accounts where we saw a spike
Whereas in the other run for a different account, we saw a dip in memory utilization. This only had 4 buckets.
